src/pages/admin/event/index.tsx

```diff
@@ -34,11 +58,83 @@ const ManageAllEventsPage: NextPage<IProps> = (props: IProps) => {
     }
   }, [allEvents, futureEvents, mode, pastEvents]);
 
+  const filteredEvents = visibleEvents
+    .filter(event => communityMatch(event, communityFilter))
+    .filter(event => event.title.toLowerCase().includes(query));
+
+  const pageSize = paginationSize === 'All' ? filteredEvents.length : paginationSize;
+
+  const nextPage = () => {
+    if ((currentPage + 1) * pageSize < filteredEvents.length) {
+      setCurrentPage(current => current + 1);
+    }
+  };
+
+  const prevPage = () => {
+    if (currentPage > 0) {
+      setCurrentPage(current => current - 1);
+    }
+  };
+
+  const changePageSize = (e: ChangeEvent<HTMLSelectElement>) => {
+    if (e.target.value === 'All') {
+      setPaginationSize(e.target.value);
+    } else {
+      setPaginationSize(parseInt(e.target.value, 10));
+    }
+  };
+
+  useEffect(() => {
+    setCurrentPage(0);
+  }, [pageSize]);
```

Comment:
I think the page number should also reset if the user changes the filter or query